jc/parsers/ipconfig.py

```diff
@@ -0,0 +1,778 @@
+        # Initialize the parsed output dictionary with all fields set to None or empty lists
+        raw_output = _parse(data)
+
+    return raw_output if raw else _process(raw_output)
+
+def _process_ipv6_address(ip_address):
+    address_split = ip_address["address"].split('%')
+    try:
+        if len(address_split) > 1:
+            address = address_split[0]
+            prefix_length = int(address_split[1])
+        else:
+            address = ip_address["address"]
+            prefix_length = None
+    except:
+        address = ip_address["address"]
+        prefix_length = None
+    return {
+              "address": address,
+              "prefix_length": prefix_length,
+              "status": ip_address["status"]
+           }
+
+def _process_ipv4_address(ip_address):
+    autoconfigured = True if ip_address.get("autoconfigured","") is not None and 'autoconfigured' in ip_address.get("autoconfigured","") else False
+    subnet_mask = ip_address["subnet_mask"]
+    return {
+              "address": ip_address["address"],
+              "subnet_mask": subnet_mask,
+              "status": ip_address["status"],
+              "autoconfigured": autoconfigured
+          }
+
+def _process(proc_data):
+    """
+    Final processing to conform to the schema.
+
+    Parameters:
+
+        proc_data:   (Dictionary) raw structured data to process
+
+    Returns:
+
+        Processed Dictionary. Structured data to conform to the schema.
+    """
+    processed = proc_data
+    
+    
+    if "ip_routing_enabled" in processed and processed["ip_routing_enabled"] is not None:
+        processed["ip_routing_enabled"] = (processed["ip_routing_enabled"].lower() == "yes")
+
+    if "wins_proxy_enabled" in processed and processed["wins_proxy_enabled"] is not None:
+        processed["wins_proxy_enabled"] = (processed["wins_proxy_enabled"].lower() == "yes")
+
+    for adapter in processed["adapters"]:
+        if "dhcp_enabled" in adapter and adapter["dhcp_enabled"] is not None:
+            adapter["dhcp_enabled"] = (adapter["dhcp_enabled"].lower() == "yes")
+        if "autoconfiguration_enabled" in adapter and adapter["autoconfiguration_enabled"] is not None:
+            adapter["autoconfiguration_enabled"] = (adapter["autoconfiguration_enabled"].lower() == "yes")
+        if "netbios_over_tcpip" in adapter and adapter["netbios_over_tcpip"] is not None:
+            adapter["netbios_over_tcpip"] = (adapter["netbios_over_tcpip"].lower() == "enabled")
+        if "lease_expires" in adapter and adapter["lease_expires"] is not None and adapter["lease_expires"] != "":
+            try:
+                adapter["lease_expires"] = datetime.strptime(adapter["lease_expires"], "%A, %B %d, %Y %I:%M:%S %p").isoformat()
+            except:
+                pass # Leave date in raw format if not parseable
+        if "lease_obtained" in adapter and adapter["lease_obtained"] is not None and adapter["lease_obtained"] != "":
+            try:
```

Comment:
I think what I'll do here is add this time format to the `jc.utils.timestamp()` class and then we can keep the original field and add epoch and iso timestamps if it parses.

Reply:
I wasn't really sure what to do here with this one. I'm not using these timestamps but I wanted to leave them in a useful and easily consumable format for others to use. ISO8601 seemed to be a good option but I have no issues moving to something consistent with other parsers for this data type.